jitsi-meet-electron: new port

takikawa opened the pull request:
#### Description

New port for jitsi-meet-electron GUI application for teleconference meetings.

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 13.5.1
Xcode 14.3.1

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. --> (no existing tickets)
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`? (no tests, new port)
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken? (no variants)

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

reneeotten commented on net/jitsi-meet-electron/Portfile:
```diff
@@ -0,0 +1,53 @@
+github.tarball_from releases
+
+name                jitsi-meet-electron
+version             2023.8.0
```
the "name" and "version" fields are automatically set from the "github.setup" line and are this redundant, please remove

reneeotten commented on net/jitsi-meet-electron/Portfile:
```diff
@@ -0,0 +1,53 @@
+version             2023.8.0
+categories          net
+installs_libs       no
+platforms           darwin
```
this is the default since MacPorts v1.7 and can this be removed

reneeotten commented on net/jitsi-meet-electron/Portfile:
```diff
@@ -0,0 +1,53 @@
+maintainers         {takikawa @takikawa} openmaintainer
+description         Jitsi Meet Electron
+long_description    Jistsi Meet Electron is a client for the open-source Jitsi Meet teleconferencing system
+homepage            https://github.com/jitsi/jitsi-meet-electron
```
this is also set automatically by the "github" PG

reneeotten commented on net/jitsi-meet-electron/Portfile:
```diff
@@ -0,0 +1,53 @@
+
+github.setup        jitsi jitsi-meet-electron 2023.8.0 v
+github.tarball_from releases
+
```
please add the  "revision 0" line

reneeotten reviewed (changes requested):
thanks for the submission @takikawa ! I've left a few general Portfile comments. I am not familiar enough with "yarn" to judge whether this is the correct approach, but seeing a comment like "fetching/building dependencies" in the "build" phase doesn't a priori look correct...

takikawa commented:
Thanks for the comments @reneeotten! I think I addressed them in the latest commit.

I also updated the version to `v2023.9.0` as a new version came out upstream since I uploaded the PR. I also changed the github `tarball_from` to the default of using tags as I think the release one didn't work for this GH repo.

> I am not familiar enough with "yarn" to judge whether this is the correct approach, but seeing a comment like "fetching/building dependencies" in the "build" phase doesn't a priori look correct...

I'm new to macports development so I'm not sure what's the right approach either, but I followed the approach used in other electron packages like these: 

  * beekeeper-studio: https://github.com/macports/macports-ports/blob/a12dee5e6049410aba5e239886986692d3b78c9a/databases/beekeeper-studio/Portfile
  * zettlr: https://github.com/macports/macports-ports/blob/a12dee5e6049410aba5e239886986692d3b78c9a/editors/zettlr/Portfile

reneeotten commented:
> I also updated the version to `v2023.9.0` as a new version came out upstream since I uploaded the PR. I also changed the github `tarball_from` to the default of using tags as I think the release one didn't work for this GH repo.

I don't see that line added (yet), but also it shouldn't be "tags", if "releases" doe not work then it should be "archive".

takikawa commented:
> I don't see that line added (yet), but also it shouldn't be "tags", if "releases" doe not work then it should be "archive".

Thanks @reneeotten, this should be updated now to use "archive" (and with updated checksums for that).

takikawa commented on net/jitsi-meet-electron/Portfile:
```diff
@@ -0,0 +1,41 @@
+}
+
+destroot {
+    copy {*}[glob ${worksrcpath}/dist/mac*/Jitsi\ Meet.app] ${destroot}${applications_dir}
```
There was a build issue on x64 (I only have an arm64 machine to test) so I switched this to glob instead of hardcode a `mac-x64` path (I guess it used `mac` on x64).

reneeotten commented on net/jitsi-meet-electron/Portfile:
```diff
@@ -0,0 +1,41 @@
+categories          net
+installs_libs       no
+license             Apache-2
+maintainers         {takikawa @takikawa} openmaintainer
```
the first entry should be an obfuscated e-mail address. The way you have specified it now implies an "@macports.org" address which you don't have.

takikawa replied:
Thanks, should be fixed now.

reneeotten commented on net/jitsi-meet-electron/Portfile:
```diff
@@ -0,0 +1,41 @@
+build.env-append    CSC_IDENTITY_AUTO_DISCOVERY=false
+
+use_configure       no
+use_xcode           yes
```
this needs a full XCode installation or is an CLT installation sufficient?

takikawa replied:
I think it doesn't need a full xcode install, so I've removed this line in the new version. It still builds so it seems ok.

pmetzger commented:
@takikawa Could you fix the last few things so we can merge this?

takikawa commented:
@pmetzger @reneeotten Sorry for the delay, was out of town recently. I believe I've fixed the remaining issues pointed out, and I've also updated the package to version 2023.10.0 as well.

pmetzger commented:
@reneeotten Are all your issues addressed?
